irohad/ordering/impl/ordering_service_impl.cpp

```diff
@@ -75,33 +78,39 @@ namespace iroha {
     }
 
     void OrderingServiceImpl::generateProposal() {
-      // TODO 05/03/2018 andrei IR-1046 Server-side shared model object
-      // factories with move semantics
-      iroha::protocol::Proposal proto_proposal;
-      proto_proposal.set_height(proposal_height_++);
-      proto_proposal.set_created_time(iroha::time::now());
       log_->info("Start proposal generation");
-      for (std::shared_ptr<shared_model::interface::Transaction> tx;
-           static_cast<size_t>(proto_proposal.transactions_size()) < max_size_
-           and queue_.try_pop(tx);) {
-        *proto_proposal.add_transactions() =
-            std::move(static_cast<shared_model::proto::Transaction *>(tx.get())
-                          ->getTransport());
-      }
-
-      auto proposal = std::make_unique<shared_model::proto::Proposal>(
-          std::move(proto_proposal));
+      std::vector<std::shared_ptr<shared_model::interface::Transaction>> txs;
```

Comment:
Declare with reserved space in advance? Pointers, after all

Reply:
For that I would need final size, which is unknown since we are pulling items from the queue

Reply:
This method should not be concurrent, so the queue size will not be changed while executing loop. So, you can reserve space as ```min(queue_size, max_size_)```